src/main/java/vg/civcraft/mc/civmodcore/CivModCorePlugin.java

```diff
@@ -1,26 +1,24 @@
 package vg.civcraft.mc.civmodcore;
 
+import java.util.logging.Logger;
+
+import org.bukkit.plugin.java.JavaPlugin;
+
 /**
  * The sole purpose of this class is to make Spigot recognize this library as a plugin and automatically
  * load the classes onto the classpath for us.
  * <p>
  * Replaces Dummy class.
  */
-public class CivModCorePlugin extends ACivMod {
-
-	@Override
-	public void onLoad() {
-		//Don't want it to load config
-	}
-
-	@Override
-	public void onEnable() {
-		//Don't want it to load commands/events
-	}
+public class CivModCorePlugin extends JavaPlugin {
 
-	@Override
-	protected String getPluginName() {
-		return "CivModCore";
+	/**
+	 * Returns the logger for this plugin. Assumes the plugin has already been loaded.
+	 *
+	 * @return The logger for this plugin.
+	 */
+	public static Logger log() {
```

Comment:
The _getPlugin()_ method gets the static singleton instance without having to add a field to the main plugin class, and you don't have to assign to a static field during startup. Also, I'm not very fond of the plugin class holding its own singleton instance, it allows you to use the plugin class for things you shouldn't be using for.

Keeping the main plugin class functionality limited to handling startup, shutdown, and the stuff that Spigot provides (logger, config) helps maintain good design. Everything else should be moved to a separate class.